### PR TITLE
[fetch] add --fetch subcommand

### PR DESCRIPTION
**Pull request by bolinfest**
[fetch] add --fetch subcommand

### CODE REVIEW

**Comment by bolinfest** [comment]
If we agree with this general direction, I can also update https://dotslash-cli.com/docs/flags/#experimental-commands as part of this PR.

File: src/subcommand.rs [hunk 1]
```diff
@@ -205,6 +213,21 @@ Learn more at {}
...
+            let dotslash_cache = DotslashCache::new();
+            let (artifact_entry, artifact_location) =
+                locate_artifact(&dotslash_data, &dotslash_cache)?;
+            if !artifact_location.executable.exists() {
+                let provider_factory = DefaultProviderFactory {};
+                download_artifact(&artifact_entry, &artifact_location, &provider_factory)?;
+                eprintln!("Downloaded {}", artifact_location.executable.display());
+            } else {
+                eprintln!("Already exists: {}", artifact_location.executable.display());
+            }
```
Comment (zertosh) [inline src/subcommand.rs:229]: How do you feel about this? Stdout always just dumps the path (for easy parsing), and stderr tells you if it was downloaded or not. Or even even just doesn't tell you if it was downloaded.

```suggestion
            if !artifact_location.executable.exists() {
                let provider_factory = DefaultProviderFactory {};
                download_artifact(&artifact_entry, &artifact_location, &provider_factory)?;
                eprintln!("Downloaded artifact");
            } else {
                eprintln!("Artifact already exists");
            }
            println!("{}", artifact_location.executable.display());
```

**Review by zertosh — CHANGES_REQUESTED** [review]

**Comment by bigfootjon** [comment]
(I'm going to hold up releasing a new version with zip support until this PR is resolved)

File: src/subcommand.rs (hunk 1 above)
Comment (zertosh) [reply src/subcommand.rs:229]: Great! I'll make these changes and land

**Pull request closed without merging**